pkg/crypto/mpc/wallet.go

```diff
@@ -128,6 +131,27 @@ func (w *Wallet) PublicKeyProto() (*secp256k1.PubKey, error) {
 	}, nil
 }
 
+func (w *Wallet) CreateEcdsaFromPublicKey() (*std_edcsa.PrivateKey, error) {
+	p, err := w.PublicKey()
+
+	// need to pad the key to 40 bytes for
+	// ecdsa key generation which sizes its buffers from
+	// 256/16 => 40
+	for len(p) < 40 {
+		p = append(p, 0)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	key, err := std_edcsa.GenerateKey(elliptic.P256(), bytes.NewReader(p))
+	if err != nil {
+		return nil, err
+	}
+
+	return key, nil
+}
```

Comment:
Creating a new keypair from an existing public key isn't secure. Anybody could recreate this key with knowledge of the public key.

Reply:
the intent is to create a key compatible with json web keys for signing this replaces the base58 public key such that signing of messages with the verification method is easier.